### PR TITLE
Drop PyTorch 1.9 support

### LINKED ISSUE #15346: Drop PyTorch 1.9 support
## Proposed refactor

See title

### Motivation

PyTorch 1.13 is released today Oct 27 (at least that's the plan).
We already support it and test it.

### Pitch

To follow our previous support window of 4 versions, drop 1.9 with PL 1.8

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

cc @carmocca @akihironitta @borda @justusschock @awaelchli @rohitgr7

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title

Fixes #15346 

### Does your PR introduce any breaking changes? If yes, please list them.

Drops PyTorch 1.9 support, as expected by our support window

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @carmocca @akihironitta @borda @justusschock @awaelchli @rohitgr7

### CODE REVIEW

**Comment by otaj** [comment]
PyTorch 1.13 is not yet out officially, so we shouldn't be dropping 1.9 yet

**Comment by carmocca** [comment]
I know. The release date is supposedly today

File: src/pytorch_lightning/README.md [hunk 1]
```diff
@@ -78,15 +78,17 @@ Lightning is rigorously tested across multiple CPUs, GPUs, TPUs, IPUs, and HPUs
...
-|     Windows py3.{7,9}      |                                                                                                                -                                                                                                                |                                                                                                                -                                                                                                                | [![Test](https://github.com/Lightning-AI/lightning/actions/workflows/ci-pytorch-tests.yml/badge.svg?branch=master&event=push)](https://github.com/Lightning-AI/lightning/actions/workflows/ci-pytorch-tests.yml) |
+# FIXME
```
Comment (carmocca) [inline src/pytorch_lightning/README.md:81]: FIXME

**Comment by carmocca** [comment]
Marking as ready to run tests. Still WIP

File: src/pytorch_lightning/CHANGELOG.md [hunk 2]
```diff
@@ -56,6 +56,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 
+- Drop PyTorch 1.8 support ([#15347](https://github.com/Lightning-AI/lightning/pull/15347))
```
Comment (otaj) [inline src/pytorch_lightning/CHANGELOG.md:59]:
```suggestion
- Drop PyTorch 1.9 support ([#15347](https://github.com/Lightning-AI/lightning/pull/15347))
```

**Review by otaj — APPROVED** [review]

**Comment by Borda** [comment]
@carmocca how much is missing here?

**Comment by carmocca** [comment]
This needs to be split into 2:
- Drop 1.9 support
- Add 1.13 testing

The latter is blocked by issues with IPUs and colossalai

**Comment by awaelchli** [comment]
@carmocca For the latter, is the issue that these libraries don't provide a package with support for 1.13, or is it our implementation that needs updates?
How about we keep the IPU job running on 1.12? And run colossalai tests only on <= 1.12? If Graphcore and Colossal don't release new packages before our next release, we can implement checks that inform the user that these strategies are not yet supported on pytorch >= 1.13.

Since these are optional dependencies for Lightning, I don't think they should block us from providing support and testing of  core for the latest pytorch version.

**Comment by carmocca** [comment]
I've removed all 1.13 related changes so that this can go before #15403.
Note that we need to update IPUs because they use 1.9

File: dockers/tpu-tests/Dockerfile [hunk 3]
```diff
@@ -1,37 +0,0 @@
-# Copyright The PyTorch Lightning team.
```
Comment (Borda) [inline dockers/tpu-tests/Dockerfile:1]: I think we need them or how is this related to dropping 1.9?

Comment (Borda) [reply dockers/tpu-tests/Dockerfile:1]: let's have it still as a separate PR :)

and I think they are used in our TPU testing: https://github.com/Lightning-AI/lightning/blob/7decc50a879c795821927709790bb515916ef99b/dockers/tpu-tests/Dockerfile#L18

Comment (carmocca) [reply dockers/tpu-tests/Dockerfile:1]: Done in https://github.com/Lightning-AI/lightning/pull/15611
